docs/paper/reference/paper-per-world-configuration.md

```diff
@@ -680,28 +679,40 @@ be found in
     and 2 is to replace all blocks with random block data.
 - max-block-height
   - **default**: 64
-  - **description**: Sets the maximum height at which anti-xray will attempt to hide ores. Only
-    multiples of 16 are allowed. Other values will be rounded down to a multiple of 16.
+  - **description**: Sets the maximum height (`y` coordinate, starting from the bottom of the world)
+    to which anti-xray will operate. Only integer multiples of 16 are accepted. All other values
+    will be rounded down. The [Minecraft Wiki page about Ore](https://minecraft.fandom.com/wiki/Ore)
+    may be helpful in determining the best value for you.
 - update-radius
   - **default**: 2
-  - **description**: Controls the distance in blocks from air or water that hidden-blocks are hidden
-    by the anti-xray engine. The maximum allowed value is 2.
+  - **description**: Radius for block updates which will be sent containing real block data when the
+    client interacts with a block. Only values `0`, `1`, and `2` are accepted. Values smaller than
+    `0` will be rounded up to `0`, while values larger than `2` will be rounded down to `2`. `0` is
+    only designed for testing purposes. Do not use it in production.
 - lava-obscures
   - **default**: false
-  - **description**: Whether or not to obfuscate blocks touching lava.
+  - **description**: Whether to obfuscate blocks touching lava. Does not work well with
+    non-stone-like ore textures.
 - use-permission
   - **default**: false
-  - **description**: Whether or not to allow players with the `paper.antixray.bypass` permission to
-    bypass anti-xray. Checking this permission is disabled by default as legacy permission plugins
-    may struggle with the number of checks made. This should only be used with modern permission
+  - **description**: Whether to allow players with the `paper.antixray.bypass` permission to bypass
+    anti-xray. Checking this permission is disabled by default as legacy permission plugins may
+    struggle with the number of checks made. This should only be used with modern permission
     plugins.
 - hidden-blocks
+
   - **default**: [copper_ore, deepslate_copper_ore, gold_ore, deepslate_gold_ore, iron_ore,
     deepslate_iron_ore, coal_ore, deepslate_coal_ore, lapis_ore, deepslate_lapis_ore,
     mossy_cobblestone, obsidian, chest, diamond_ore, deepslate_diamond_ore, redstone_ore,
     deepslate_redstone_ore, clay, emerald_ore, deepslate_emerald_ore, ender_chest]
-  - **description**: List of blocks to be hidden in engine mode
-  - **note**: This list is using Mojang server names, _not_ bukkit names.
+  - **description**: With `engine-mode: 1`, these blocks will be replaced by `replacement-blocks`.
+    Air blocks are not allowed here in `engine-mode: 1`.
+
+    With `engine-mode: 2`, these blocks will be randomly placed in the world, replacing
+    `replacement-blocks`. Tile entities are not allowed here in `engine-mode: 2`. Individual blocks
```

Comment:
Not only `replacement-blocks` are replaced, but also the blocks in this list (`hidden-blocks`) are replaced (see original gist).